File: 0001_TwoSum/python/main.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        my_map = {}

        for i, n in enumerate(nums):
            if target - n in my_map:
                return[my_map[target - n],i]
            my_map[n] = i

File: 0001_TwoSum/cpp/main.cpp
#include <unordered_map>
class Solution {
public:
  vector<int> twoSum(vector<int> &nums, int target) {
    std::unordered_map<int, int> hashmap;

    for (int i = 0; i < nums.size(); i++) {
      int component = target - nums[i];

      if (hashmap.find(component) != hashmap.end()) {
        return {hashmap[component], i};
      }
      hashmap[nums[i]] = i;
    }
    return {};
  }
};
